# uoc_GWAS
Code used during the realization of the project GWAS from the UOC-UB

The scripts are:

remove_missing.py: Delete missing genotypes from hapMap file.
density.py: Filter by density a genotypes file from hapMap to perfom simulations.
mask.py: Mask a X percentage of genotypes from hapMap to perfom simulations.
impute2.py: Run IMPUTE2 analysis automatically for the whole chromosome with intervals of 5 million of bases.

**possible improvements:
beagle.py
mach.py
plink.py
....
